<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from urdf/carModel.xacro            | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="car_model" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <material name="gray">
    <color rgba="0.3 0.3 0.3 1"/>
  </material>
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.25 0.5 0.0625"/>
      </geometry>
      <material name="black"/>
      <origin rpy="0 0 0" xyz="0 0 0.125"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.25 0.5 0.0625"/>
        <origin rpy="0 0 0" xyz="0 0 0.125"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="5"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <link name="front_axle">
    <visual>
      <geometry>
        <cylinder length="0.4" radius="0.0625"/>
      </geometry>
      <material name="gray"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.4" radius="0.0625"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <joint name="front_axle_to_base" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-0.785" upper="0.785" velocity="1.0"/>
    <parent link="base_link"/>
    <child link="front_axle"/>
    <origin rpy="0 -1.57 0" xyz="0 0.3 0.125"/>
  </joint>
  <!--
                <xacro:global_joint child="${frontback}_axle"/>
        -->
  <link name="back_axle">
    <visual>
      <geometry>
        <cylinder length="0.4" radius="0.0625"/>
      </geometry>
      <material name="gray"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.4" radius="0.0625"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <joint name="back_axle_to_base" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-0.785" upper="0.785" velocity="1.0"/>
    <parent link="base_link"/>
    <child link="back_axle"/>
    <origin rpy="0 -1.57 0" xyz="0 -0.3 0.125"/>
  </joint>
  <!--
                <xacro:global_joint child="${frontback}_axle"/>
        -->
  <link name="front_left_wheel">
    <visual>
      <geometry>
        <cylinder length="0.125" radius="0.125"/>
      </geometry>
      <material name="white"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.125" radius="0.125"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="20"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <joint name="front_left_wheel_to_front_axle" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-0.393" upper="0.785" velocity="1.0"/>
    <parent link="base_link"/>
    <child link="front_left_wheel"/>
    <origin rpy="0 -1.57 0" xyz="-0.25 0.3 0.125"/>
  </joint>
  <!--
                <xacro:global_joint child="front_${leftright}_wheel" />
                -->
  <link name="front_right_wheel">
    <visual>
      <geometry>
        <cylinder length="0.125" radius="0.125"/>
      </geometry>
      <material name="white"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.125" radius="0.125"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="20"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <joint name="front_right_wheel_to_front_axle" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-0.785" upper="0.393" velocity="1.0"/>
    <parent link="base_link"/>
    <child link="front_right_wheel"/>
    <origin rpy="0 -1.57 0" xyz="0.25 0.3 0.125"/>
  </joint>
  <!--
                <xacro:global_joint child="front_${leftright}_wheel" />
                -->
  <link name="back_left_wheel">
    <visual>
      <geometry>
        <cylinder length="0.125" radius="0.125"/>
      </geometry>
      <material name="white"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.125" radius="0.125"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="20"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <joint name="back_left_wheel_to_back_axle" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="3.14" upper="3.14" velocity="1.0"/>
    <parent link="base_link"/>
    <child link="back_left_wheel"/>
    <origin rpy="0 -1.57 0" xyz="-0.25 -0.3 0.125"/>
  </joint>
  <!--
                <xacro:global_joint child="back_${leftright}_wheel"/>
        -->
  <link name="back_right_wheel">
    <visual>
      <geometry>
        <cylinder length="0.125" radius="0.125"/>
      </geometry>
      <material name="white"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.125" radius="0.125"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="20"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <joint name="back_right_wheel_to_back_axle" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="1.0"/>
    <parent link="base_link"/>
    <child link="back_right_wheel"/>
    <origin rpy="0 -1.57 0" xyz="0.25 -0.3 0.125"/>
  </joint>
  <!--
                <xacro:global_joint child="back_${leftright}_wheel"/>
        -->
  <!--
        -->
</robot>

